pkg/apis/serving/v1/revision_helpers.go

```diff
@@ -164,6 +164,7 @@ func (r *Revision) GetRoutingStateModified() time.Time {
 // IsReachable returns whether or not the revision can be reached by a route.
 func (r *Revision) IsReachable() bool {
 	return r.Labels[serving.RouteLabelKey] != "" ||
+		r.Annotations[serving.RoutesAnnotationKey] != "" ||
 		RoutingState(r.Labels[serving.RoutingStateLabelKey]) == RoutingStateActive
```

Comment:
serving.RoutingStateLabelKey is the intended label here to determine reachability. The RoutesAnnotationKey is metadata we use to determine the value for it, but in the future we may have states where a Route is referenced, but we don't intend as active.

I'd leave this as-is.

Reply:
It does not seem to be conclusive though — since tests obviously flake.
 Revision referenced, you mean?

Reply:
revision referenced by route I mean. This should be conclusive and the two annotations are currently set together.

Rather than change this we might investigate if a revision is created without the proper annotation or the labeler isn't applying it. I wouldn't want to mask an underlying problem. Do we have logs showing this?